src/java.base/share/classes/java/util/Collections.java

```diff
@@ -2737,41 +2752,97 @@ public void add(int index, E element) {
                                             mutex);
             }
         }
-
-        @Override
         public void replaceAll(UnaryOperator<E> operator) {
             synchronized (mutex) {list.replaceAll(operator);}
         }
-        @Override
         public void sort(Comparator<? super E> c) {
             synchronized (mutex) {list.sort(c);}
         }
+        public void addFirst(E element) {
+            synchronized (mutex) {list.addFirst(element);}
+        }
+        public void addLast(E element) {
+            synchronized (mutex) {list.addLast(element);}
+        }
+        public E getFirst() {
+            synchronized (mutex) {return list.getFirst();}
+        }
+        public E getLast() {
+            synchronized (mutex) {return list.getLast();}
+        }
+        public E removeFirst() {
+            synchronized (mutex) {return list.removeFirst();}
+        }
+        public E removeLast() {
+            synchronized (mutex) {return list.removeLast();}
+        }
+
+        /**
+         * Reversed view handling. The reversedView field is transient
+         * and is initialized to null upon construction of the wrapper
+         * and upon deserialization. Reversed views are not serializable.
+         * The reversed view is created on first call to the reversed()
+         * method.
+         *
+         * There are four objects at play here:
+         *
+         * L = original list
+         * Lr = reversed view of original list
+         * S = synchronized forward wrapper: its backing list is L
+         * Sr = synchronized reversed wrapper: its backing list is Lr
+         *
+         * The reversedView field of S points to Sr and vice versa.
+         * This enables the reversed() method always to return the same
+         * object, and for S.reversed().reversed() to return S. This isn't
+         * strictly necessary, because all internal locking is done on S
+         * (which is passed around as mutex). But in the case where the
+         * client does external locking, it's good to minimize the number
+         * of different instances. Note however that external locking on
+         * the reversed wrapper Sr can't be made to work properly with
+         * internal or external locking on the forward wrapper S. (Sublists
+         * of a synchronized wrapper, reversed or not, have the same issue.)
+         *
+         * An alternative would be to have a ReversedSynchronizedList view class. However,
+         * that would require two extra classes (one RandomAccess and one not) and it would
+         * complicate the serialization story.
+         */
+        public List<E> reversed() {
+            synchronized (mutex) {
+                if (reversedView == null) {
+                    reversedView = new SynchronizedList<>(list.reversed(), mutex, this);
+                }
+                return reversedView;
+            }
+        }
```

Comment:
`SynchronizedRandomAccessList` overrides `subList`, should it also override `reversed`?